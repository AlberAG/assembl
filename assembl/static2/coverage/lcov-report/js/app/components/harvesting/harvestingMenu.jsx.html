<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/harvesting/harvestingMenu.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">js/app/components/harvesting</a> harvestingMenu.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.95% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.92% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/13</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import addPostExtractMutation from '../../graphql/mutations/addPostExtract.graphql'; // eslint-disable-line
import updateExtractMutation from '../../graphql/mutations/updateExtract.graphql'; // eslint-disable-line
import deleteExtractMutation from '../../graphql/mutations/deleteExtract.graphql'; // eslint-disable-line
import HarvestingAnchor from './harvestingAnchor';
import HarvestingBox from './harvestingBox';
import HarvestingBadge from './harvestingBadge';
&nbsp;
type Props = {
  extracts: Array&lt;Extract&gt;,
  postId: string,
  lang: string,
  displayHarvestingBox: boolean,
  displayHarvestingAnchor: boolean,
  harvestingAnchorPosition: Object,
  setHarvestingBoxDisplay: Function,
  handleClickAnchor: Function,
  cancelHarvesting: Function,
  refetchPost: Function,
  isAuthorAccountDeleted: boolean,
  showNuggetAction: boolean
};
&nbsp;
type State = {
  displayExtractsBox: boolean
};
&nbsp;
class HarvestingMenu extends React.Component&lt;Props, State&gt; {
  constructor(props: Props) {
    super(props);
    this.state = {
      displayExtractsBox: false
    };
  }
&nbsp;
  handleMouseDown = <span class="fstat-no" title="function not covered" >(e</span>vent: SyntheticMouseEvent&lt;&gt;) =&gt; {
    // This would otherwise clear the selection
<span class="cstat-no" title="statement not covered" >    event.preventDefault();</span>
<span class="cstat-no" title="statement not covered" >    return false;</span>
  };
&nbsp;
  toggleExtractsBox = <span class="fstat-no" title="function not covered" >()</span>: void =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >pr</span>evState =&gt; (<span class="cstat-no" title="statement not covered" >{ displayExtractsBox: !prevState.displayExtractsBox })</span>);</span>
  };
&nbsp;
  render() {
    const {
      setHarvestingBoxDisplay,
      displayHarvestingBox,
      displayHarvestingAnchor,
      postId,
      handleClickAnchor,
      extracts,
      harvestingAnchorPosition,
      refetchPost,
      cancelHarvesting,
      isAuthorAccountDeleted,
      showNuggetAction,
      lang
    } = this.props;
    const selection = window.getSelection();
    const { displayExtractsBox } = this.state;
    const showHarvestingBadge = !displayExtractsBox &amp;&amp; extracts &amp;&amp; extracts.length &gt; 0;
    const showBoxWithExtracts = displayExtractsBox &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >extracts </span>&amp;&amp; <span class="branch-2 cbranch-no" title="branch not covered" >extracts.length &gt; 0 </span>&amp;&amp; <span class="branch-3 cbranch-no" title="branch not covered" >!displayHarvestingBox;</span>
    const showBoxInHarvestingMode = displayHarvestingBox &amp;&amp; selection.toString().length &gt; 0;
    const showHarvestingAnchor = displayHarvestingAnchor &amp;&amp; selection.toString().length &gt; 0;
    return (
      &lt;div&gt;
        {showHarvestingBadge &amp;&amp; &lt;HarvestingBadge toggleExtractsBox={this.toggleExtractsBox} extractsNumber={extracts.length} /&gt;}
        &lt;div className="harvesting-container"&gt;
          {showBoxWithExtracts &amp;&amp; (
<span class="branch-1 cbranch-no" title="branch not covered" >            &lt;HarvestingBox</span>
              postId={postId}
              key={`extracts-${postId}`}
              extracts={extracts}
              isAuthorAccountDeleted={isAuthorAccountDeleted}
              displayHarvestingBox={displayHarvestingBox}
              refetchPost={refetchPost}
              cancelHarvesting={cancelHarvesting}
              setHarvestingBoxDisplay={setHarvestingBoxDisplay}
              showNuggetAction={showNuggetAction}
              toggleExtractsBox={this.toggleExtractsBox}
            /&gt;
          )}
          {showBoxInHarvestingMode &amp;&amp; (
            &lt;HarvestingBox
              postId={postId}
              key={`harvesting-${postId}`}
              selection={selection}
              lang={lang}
              displayHarvestingBox={displayHarvestingBox}
              refetchPost={refetchPost}
              cancelHarvesting={cancelHarvesting}
              setHarvestingBoxDisplay={setHarvestingBoxDisplay}
              showNuggetAction={showNuggetAction}
            /&gt;
          )}
          {showHarvestingAnchor &amp;&amp; (
            &lt;HarvestingAnchor
              displayHarvestingBox={displayHarvestingBox}
              handleMouseDown={this.handleMouseDown}
              anchorPosition={harvestingAnchorPosition}
              handleClickAnchor={handleClickAnchor}
            /&gt;
          )}
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
export default HarvestingMenu;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
