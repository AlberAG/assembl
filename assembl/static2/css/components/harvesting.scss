.harvesting-anchor {
  position: absolute;
  left: 250px;
}

.left-side-harvesting-button {
  &:hover {
    .assembl-icon-catch:before {
      font-weight: bold;
    }

    .left-side-harvesting-button__label {
      visibility: visible;
    }

    .left-side-harvesting-button__button__inside {
      background-color: #ffed5b;
    }
  }

  .left-side-harvesting-button__label {
    position: absolute;
    left: -100px;
    top: 0;
    width: 100px;
    margin-top: 11px;
    padding-right: 9px;
    text-align: right;
    visibility: hidden;
    font-size: 12px;
    opacity: 0.55;
    font-weight: bold;
    letter-spacing: 0.1em;
  }

  .left-side-harvesting-button__button {
    background: white;
    border: 1px solid #ccc;
    border-radius: 50%;
    width: 44px;
    height: 44px;
    box-shadow: 2px 2px 4px 0 #ccc;
    cursor: pointer;

    .left-side-harvesting-button__button__inside {
      text-align: center;
      margin: 4px;
      width: calc(100% - 8px);
      height: calc(100% - 8px);
      background-color: #fff7b5;
      border-radius: 50%;
      display: flex;
    }
  }
}

.confirm-harvest-button {
  text-align: center;
  padding: 1px 0 0 6px;
  font-size: 20px;

  &:before {
    margin-right: 2px;
    font-size: $font-size-sm;
  }
}

.extracts.extracts--is-harvesting {
  left: 0;
  max-width: 355px;
  width: 20vw;
  line-height: initial;

  .extract {
    border-left: $border-theme;
    border-bottom: 1px solid #ddd;
    border-right: 1px solid #ddd;
    border-top: 1px solid #ddd;
    background-color: $white;

    .extract-txt {
      quotes: "«" "»";
      padding: 20px;

      &:before {
        content: open-quote;
        padding-right: 5px;
      }

      &:after {
        content: close-quote;
        padding-left: 5px;
      }
    }

    .extract-header {
      padding: 25px;
      border-bottom: 1px solid #ccc;

      .extract-harvester {
        display: flex;

        .extract-harvester-right {
          padding-left: 10px;
        }
      }
    }
  }
}

.app.harvesting-mode-on {
  .post-body--is-harvestable {
    /* FIXME: Factorizing these ::selection rules cancels effect on Firefox */
    .post-body-title {
      ::selection {
        background: $harvesting-yellow;
      }

      ::-moz-selection {
        background: $harvesting-yellow;
      }
    }

    .post-body-content > div::selection {
      background: $harvesting-yellow;
    }

    .post-body-content > div::-moz-selection {
      background: $harvesting-yellow;
    }

    .post-body-content > div > p::selection {
      background: $harvesting-yellow;
    }

    .post-body-content > div > p::-moz-selection {
      background: $harvesting-yellow;
    }
  }
}

.active-box {
  border-left-color: $harvesting-green;
}

.harvesting-box {
  position: absolute;
  box-shadow: 2px 2px 4px 0 $grey1;
  left: 0;
  z-index: 99;
  width: 310px;

  .harvesting-box-header {
    padding: 20px;
    border-bottom: 1px solid $grey1;
  }

  .harvesting-status {
    display: inline-block;
    font-size: $font-size-xs;

    .harvesting-in-progress {
      color: $black;
      margin-left: -12px;
      padding: 0 0 15px;
    }

    .validated-harvesting {
      color: $harvesting-green;
      margin-left: -12px;
      padding: 0 0 15px;
    }

    .harvesting-status-label {
      display: inline-block;
      vertical-align: bottom;
    }
  }

  .profile {
    font-size: $font-size-xs;
    color: $grey2;
    margin-left: -8px;

    img {
      border-radius: 50%;
      margin: 0 10px 0 6px;
    }

    .assembl-icon-profil {
      vertical-align: bottom;

      &:before {
        font-size: 30px;
        margin-right: 10px;
      }
    }

    .harvesting-date {
      font-size: $font-size-xxs;
    }

    .harvesting-infos {
      display: inline-block;
      vertical-align: bottom;
    }
  }

  .button-bar {
    padding-top: 5px;
    float: right;
    margin-right: -12px;

    .assembl-icon-ellipsis-vert {
      &:before {
        font-size: 20px;
        vertical-align: middle;
        padding: 2px 0 0;
      }
    }

    .btn[disabled] {
      opacity: 1;
      border-color: $grey5;

      span {
        color: $grey5;
      }

      &:hover {
        background-color: $white;
        border-color: $grey5;

        span {
          color: $grey5;
        }
      }
    }

    .btn.active {
      background-color: $white;
      border-color: $second-color;
      box-shadow: none;

      span {
        color: $second-color;
      }
    }

    .btn-default {
      margin: 0 3px;
      border-radius: 5px;
      width: 26px;
      height: 26px;
      border-color: $grey2;

      span {
        color: $grey2;
      }

      &:hover {
        background-color: $white;
        border-color: $second-color;

        span {
          color: $second-color;
        }
      }

      &:focus {
        background-color: $white;
        border-color: $second-color;
        outline: none;

        span {
          color: $second-color;
        }
      }

      span {
        &:before {
          font-size: 12px;
          display: block;
          margin: 0 0 0 -6px;
        }
      }

      .assembl-icon-check {
        &:before {
          font-size: 15px;
        }
      }
    }
  }

  .harvesting-box-body {
    padding: 20px;
    font-size: $font-size-xs;
    color: $grey2;
  }

  .harvesting-box-footer {
    text-align: center;
    padding: 20px 0;
    background-color: $grey1;

    .btn-default {
      margin: 0 10px;
    }
  }
}

.extract-in-message {
  display: inline;
}

.is-harvesting {
  .extract-in-message {
    background-color: $harvesting-green-light;
  }
}
