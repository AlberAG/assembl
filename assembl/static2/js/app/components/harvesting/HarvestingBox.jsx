// @flow
import React from 'react';
import { Button } from 'react-bootstrap';
import { Translate } from 'react-redux-i18n';

type Props = {
  positionX: number,
  positionY: number,
  selection: string
};

type State = {
  disabled: boolean
};

class HarvestingBox extends React.Component<void, Props, State> {
  props: Props;

  state: State;

  constructor(props: Props) {
    super(props);
    this.state = {
      disabled: true
    };
  }

  validateHarvesting = (): void => {
    this.setState({ disabled: false });
  };

  cancelHarvesting = (): void => {
    this.setState({ disabled: true });
  };

  render() {
    const { positionX, positionY, selection } = this.props;
    const { disabled } = this.state;
    return (
      <div style={{ top: positionY, left: positionX }} className="theme-box harvesting-box">
        <div className="harvesting-box-header">
          <div className="profil">
            <span className="assembl-icon-profil grey" />
            <span className="username">Pauline Thomas</span>
          </div>
          <div className="button-bar">
            <Button disabled={disabled}>
              <span className="assembl-icon-check grey" />
            </Button>
            <Button disabled={disabled}>
              <span className="assembl-icon-edit grey" />
            </Button>
            <Button disabled={disabled}>
              <span className="assembl-icon-delete grey" />
            </Button>
            <Button disabled={disabled}>
              <span className="assembl-icon-pepite grey" />
            </Button>
          </div>
        </div>
        <div className="harvesting-box-body">{selection}</div>
        <div className="harvesting-box-footer">
          <Button className="button-submit button-dark" onClick={this.validateHarvesting}>
            <Translate value="common.attachFileForm.submit" />
          </Button>
          <Button className="button-cancel button-dark" onClick={this.cancelHarvesting}>
            <Translate value="debate.confirmDeletionButtonCancel" />
          </Button>
        </div>
      </div>
    );
  }
}

export default HarvestingBox;