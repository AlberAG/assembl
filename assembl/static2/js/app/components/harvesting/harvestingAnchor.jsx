// @flow
import React from 'react';
import { OverlayTrigger } from 'react-bootstrap';
import { ANCHOR_SIZE } from '../../constants';
import { harvestingTooltip } from '../common/tooltips';

type Props = {
  anchorPosition: Object,
  displayHarvestingBox: Function,
  handleMouseDown: Function
};

const HarvestingAnchor = ({ displayHarvestingBox, handleMouseDown, anchorPosition }: Props) => (
  <OverlayTrigger placement="top" overlay={harvestingTooltip}>
    <div
      className="harvesting-anchor"
      style={{ top: `${anchorPosition.y - ANCHOR_SIZE}px`, left: `${anchorPosition.x - ANCHOR_SIZE}px` }}
    >
      <div className="left-side-harvesting-button">
        <div className="left-side-harvesting-button__button" onClick={displayHarvestingBox} onMouseDown={handleMouseDown}>
          <div className="left-side-harvesting-button__button__inside">
            <span className="confirm-harvest-button assembl-icon-catch" />
          </div>
        </div>
      </div>
    </div>
  </OverlayTrigger>
);

export default HarvestingAnchor;