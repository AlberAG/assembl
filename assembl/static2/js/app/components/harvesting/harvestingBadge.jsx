// @flow
import * as React from 'react';

type Props = {
  setExtractsBoxDisplay: Function,
  extractsNumber: number
};

const harvestingBadge = ({ setExtractsBoxDisplay, extractsNumber }: Props) => (
  <div className="harvesting-anchor" onClick={setExtractsBoxDisplay}>
    <div>{extractsNumber}</div>
    <div className="left-side-harvesting-button">
      <div className="left-side-harvesting-button__button">
        <div className="left-side-harvesting-button__button__inside">
          <span className="confirm-harvest-button assembl-icon-catch" />
        </div>
      </div>
    </div>
  </div>
);

export default harvestingBadge;