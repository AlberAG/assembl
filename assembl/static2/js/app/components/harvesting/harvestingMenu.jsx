// @flow
import * as React from 'react';
import addPostExtractMutation from '../../graphql/mutations/addPostExtract.graphql'; // eslint-disable-line
import updateExtractMutation from '../../graphql/mutations/updateExtract.graphql'; // eslint-disable-line
import deleteExtractMutation from '../../graphql/mutations/deleteExtract.graphql'; // eslint-disable-line
import HarvestingAnchor from './harvestingAnchor';
import HarvestingBox from './harvestingBox';
import HarvestingBadge from './harvestingBadge';

type Props = {
  extracts: Array<Extract>,
  postId: string,
  lang: string,
  displayHarvestingBox: boolean,
  displayHarvestingAnchor: boolean,
  harvestingAnchorPosition: Object,
  setHarvestingBoxDisplay: Function,
  handleClickAnchor: Function,
  cancelHarvesting: Function,
  refetchPost: Function,
  isAuthorAccountDeleted: boolean,
  showNuggetAction: boolean
};

type State = {
  displayExtractsBox: boolean
};

class HarvestingMenu extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      displayExtractsBox: false
    };
  }

  handleMouseDown = (event: SyntheticMouseEvent<>) => {
    // This would otherwise clear the selection
    event.preventDefault();
    return false;
  };

  setExtractsBoxDisplay = (): void => {
    this.setState(prevState => ({ displayExtractsBox: !prevState.displayExtractsBox }));
  };

  render() {
    const {
      setHarvestingBoxDisplay,
      displayHarvestingBox,
      displayHarvestingAnchor,
      postId,
      handleClickAnchor,
      extracts,
      harvestingAnchorPosition,
      refetchPost,
      cancelHarvesting,
      isAuthorAccountDeleted,
      showNuggetAction,
      lang
    } = this.props;
    const selection = window.getSelection();
    const { displayExtractsBox } = this.state;
    return (
      <div className="harvesting-container">
        {displayExtractsBox && extracts && extracts.length > 0 ? (
          <HarvestingBox
            postId={postId}
            key={`extracts-${postId}`}
            extracts={extracts}
            isAuthorAccountDeleted={isAuthorAccountDeleted}
            displayHarvestingBox={displayHarvestingBox}
            harvestingBoxPosition={null}
            refetchPost={refetchPost}
            cancelHarvesting={cancelHarvesting}
            setHarvestingBoxDisplay={setHarvestingBoxDisplay}
            showNuggetAction={showNuggetAction}
            setExtractsBoxDisplay={this.setExtractsBoxDisplay}
          />
        ) : null}
        {!displayExtractsBox && (
          <HarvestingBadge setExtractsBoxDisplay={this.setExtractsBoxDisplay} extractsNumber={extracts.length} />
        )}
        {displayHarvestingBox && (
          <HarvestingBox
            postId={postId}
            selection={selection}
            lang={lang}
            extract={null}
            displayHarvestingBox={displayHarvestingBox}
            refetchPost={refetchPost}
            cancelHarvesting={cancelHarvesting}
            setHarvestingBoxDisplay={setHarvestingBoxDisplay}
            showNuggetAction={showNuggetAction}
          />
        )}
        {displayHarvestingAnchor &&
          selection.toString().length > 0 && (
            <HarvestingAnchor
              displayHarvestingBox={displayHarvestingBox}
              handleMouseDown={this.handleMouseDown}
              anchorPosition={harvestingAnchorPosition}
              handleClickAnchor={handleClickAnchor}
            />
          )}
      </div>
    );
  }
}

export default HarvestingMenu;