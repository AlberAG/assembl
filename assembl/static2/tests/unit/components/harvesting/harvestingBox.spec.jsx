import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DumbHarvestingBox } from '../../../../js/app/components/harvesting/harvestingBox';
import * as fakeData from './fakeData';

describe('harvestingBox component', () => {
  const setExtractsBoxDisplaySpy = jest.fn(() => {});
  it('should match harvestingBox snapshot', () => {
    const { extract } = fakeData;
    const props = {
      postId: '12345',
      extract: extract,
      index: 0,
      previousExtractId: '876876876',
      harvestingAnchorPosition: { x: 100, y: 200 },
      contentLocale: 'fr',
      harvestingDate: 'il y a 5 jours',
      showNuggetAction: true,
      setExtractsBoxDisplay: setExtractsBoxDisplaySpy
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbHarvestingBox {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('should match submitted harvestingBox snapshot', () => {
    const { submittedExtract } = fakeData;
    const props = {
      postId: '12345',
      extract: submittedExtract,
      index: 0,
      previousExtractId: '876876876',
      harvestingAnchorPosition: { x: 100, y: 200 },
      contentLocale: 'fr',
      harvestingDate: 'il y a 5 jours',
      showNuggetAction: true,
      setExtractsBoxDisplay: setExtractsBoxDisplaySpy
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbHarvestingBox {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('should match submitted harvestingBox without nugget action snapshot', () => {
    const { submittedExtract } = fakeData;
    const props = {
      postId: '12345',
      extract: submittedExtract,
      index: 0,
      previousExtractId: '876876876',
      harvestingAnchorPosition: { x: 100, y: 200 },
      contentLocale: 'fr',
      harvestingDate: 'il y a 5 jours',
      showNuggetAction: false,
      setExtractsBoxDisplay: setExtractsBoxDisplaySpy
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbHarvestingBox {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});