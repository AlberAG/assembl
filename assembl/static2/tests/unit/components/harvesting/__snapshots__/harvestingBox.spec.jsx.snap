// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`harvestingBox component should match harvestingBox snapshot 1`] = `
<div
  className="theme-box harvesting-box active-box"
  id="harvesting-box-987643"
>
  <div
    className="harvesting-box-header"
  >
    <div
      className="harvesting-status"
    >
      <div
        className="validated-harvesting"
      >
        <span
          className="confirm-harvest-button assembl-icon-catch"
        />
        <div
          className="harvesting-status-label"
        >
          <div>
            <Translate
              value="harvesting.harvestSelection"
            />
          </div>
          <div>
            <Translate
              value="harvesting.validated"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="button-bar"
    >
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="validateExtractTooltip"
            placement="right"
          >
            <Translate
              value="harvesting.validateExtract"
            />
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="active"
          disabled={false}
        >
          <span
            className="assembl-icon-check grey"
          />
        </Button>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="editExtractTooltip"
            placement="right"
          >
            <Translate
              value="harvesting.editExtract"
            />
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className=""
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="assembl-icon-edit grey"
          />
        </Button>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="deleteExtractTooltip"
            placement="right"
          >
            <Translate
              value="harvesting.deleteExtract"
            />
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="assembl-icon-delete grey"
          />
        </Button>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="nuggetExtractTooltip"
            placement="right"
          >
            <Translate
              value="harvesting.nuggetExtract"
            />
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className=""
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="assembl-icon-pepite grey"
          />
        </Button>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="qualifyExtractTooltip"
            placement="right"
          >
            <Translate
              value="harvesting.qualifyExtract"
            />
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <span
          className="assembl-icon-ellipsis-vert grey"
        />
      </OverlayTrigger>
    </div>
    <div
      className="profile"
    >
      <AvatarImage
        userId={31}
        userName="John Doe"
      />
      <div
        className="harvesting-infos"
      >
        <div
          className="username"
        >
          John Doe
        </div>
        <div
          className="harvesting-date"
          title="2018-03-29T16:28:27.324276+00:00"
        >
          il y a 5 jours
        </div>
      </div>
    </div>
  </div>
  <div
    className="harvesting-box-body"
  >
    <div>
      Hello world!
    </div>
  </div>
</div>
`;
